src/base/bittorrent/tracker.h

```diff
@@ -51,28 +51,33 @@ namespace BitTorrent
 {
     struct Peer
```

Comment:
>I guess I didn't quite get what you mean by moving pure private thing to .cpp. I would appreciate if you could clarify this matter.

Peer and TrackerAnnounceRequest structs are used internally only so, as I said before, move its declarations to .cpp (not use anonymous namespace for it)  and use forward declarations here (outside BitTorrent namespace):

```c++
struct Peer;
struct TrackerAnnounceRequest;

typedef QHash<QString, Peer> PeerList;
typedef QHash<QString, PeerList> TorrentList;

namespace BitTorrent
{
```

Also move Peer methods definitions in struct itself (no need to separate it since we have it in .cpp).